src/plugin/plugin_manager.js

```diff
@@ -174,10 +174,23 @@ export function getLatestVersionNumber() {
 }
 
 export function isUpdateAvailable() {
-  if (!pluginObject) return;
-  let current = pluginObject.VERSION;
-  let latest  = getLatestVersionNumber();
-  return utils.versionCompare(latest, current) > 0;
+  return new Promise ((resolve, reject) =>{
+    callWhenPluginReady(() => {
+      if (pluginObject) {
+        var current = pluginObject.VERSION;
+        var latest = getLatestVersionNumber();
+        utils.versionCompare(latest, current) > 0 ? resolve(true) : resolve(false);
+      } else {
+        reject("plugin object is not available");
+      }
+    });
+  });
+}
+
+export function checkForUpdate() {
+  if (isUpdateAvailable() && config.autoUpdate) {
```

Comment:
Better to change the order of the conditions; Do the config.autoUpdate check first and then do the heavy isUpdateAvailable() check (which involves http GET).

Reply:
function removed since no longer use